src/QuLDE.jl

```diff
@@ -0,0 +1,89 @@
+
+struct QuLDEParam{W,L,Q}
+    k::Int
+    t::L
+    C_tilda::W
+    D_tilda::W
+    N::W
+    V::GeneralMatrixBlock{1}
```

Comment:
This is not correct here, the type parameter of GeneralMatrixBlock is

```julia
"""
    GeneralMatrixBlock{M, N, MT} <: PrimitiveBlock{N}
General matrix gate wraps a matrix operator to quantum gates. This is the most
general form of a quantum gate. `M` is the hilbert dimension (first dimension),
`N` is the hilbert dimension (second dimension) of current quantum state. For
most quantum gates, we have ``M = N``.
"""
struct GeneralMatrixBlock{M, N, T, MT <: AbstractMatrix{T}} <: PrimitiveBlock{N}
```

If you don't want to write these type parameters just write it in this way:

```
struct QuLDEParam{W,L,Q, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixBlock{Q}}
    k::Int
    t::L
    C_tilda::W
    D_tilda::W
    N::W
    V::VType
   VS1::SType
    VS2::SType
    WS1::SType
    WS2::SType
end
```

Reply:
I used the second thing you suggested.